crates/build/src/manifest.rs

```diff
@@ -1,12 +1,43 @@
-use std::path::PathBuf;
-
 use serde::{Deserialize, Serialize};
+use spin_app::locked::FixedVersion;
+use std::path::PathBuf;
 
 #[derive(Clone, Debug, Deserialize, Serialize)]
-#[serde(tag = "spin_version")]
+#[serde(untagged)]
 pub(crate) enum BuildAppInfoAnyVersion {
-    #[serde(rename = "1")]
-    V1(BuildAppInfoV1),
+    V1Old {
+        #[serde(deserialize_with = "deserialize_fixed_version_1")]
```

Comment:
I notice the enum is marked as Serialize (though I'm not sure we ever serialise it!) but we only customise the Deserialize - does Serialize Just Work?

Reply:
It would "work" but serialize the version as an int, which isn't right. I'm wondering if `FixedVersion` shouldn't be reused from `spin_app` and instead copied & adapted to be a string value.

Reply:
Ahh good catch. I'd vote for just dropping the Serialize.

Reply:
I think for backwards compat of `LockedApp`, we need `FixedVersion` to continue to contain a `usize` instead of string.

Reply:
Created a `FixedStringVersion` which deserializes from `String` into a `usize` constant. I duplicated this across `manifest` and `config` instead of bringing in `spin-app`. I also removed `Serialize` since it is not used.